modules/core/shared/src/main/scala/gem/config/DynamicConfig.scala

```diff
@@ -83,6 +84,17 @@ object DynamicConfig {
     type GmosNorthDefinition = GmosDefinition[GmosNorthDisperser, GmosNorthFilter, GmosNorthFpu]
     type GmosSouthDefinition = GmosDefinition[GmosSouthDisperser, GmosSouthFilter, GmosSouthFpu]
 
+    final case class Gnirs(
+      camera:    GnirsCamera,
+      disperser: GnirsDisperser,
+      filter:    GnirsFilter,
+      fpu:       Either[GnirsFpuOther, GnirsFpuSlit],
+      prism:     GnirsPrism,
+      // TODO: static smart
+      wellDepth: GnirsWellDepth
+    ) extends SmartGcalSearchKey with SmartGcalDefinitionKey
```

Comment:
I'm a bit confused because this doesn't seem to map very well to what I see in the [old codebase](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/calunit/smartgcal/keys/ConfigKeyGnirs.java#L36-L41).  There is no camera but there is an imaging vs. spectroscopy mode, filter shouldn't be there but cross dispersed mode should be, etc.  Am I looking at it wrong?

Reply:
Those correspond to some of the GNIRS changes that were relatively easy to implement in the enumerations. These were discussed with @andrewwstephens, ideally there should be more changes, but the remaining changes were trickier to implement.

For example, Pixel Scale is now a column on the camera table, since the camera the choice of camera dictates the pixel scale.

I believe imaging vs. spectroscopy is distinguished now by the FPU.

Reply:
The mapping from the old keys to the new one isn't clear to me.  Before it was imaging vs. spectroscopy.  How will you go from "SPECTROSCOPY" in the configuration file to a specific FPU?  Will the smart gcal config now differ depending upon which of the spectroscopy FPU options is selected?  How will you set the filter if it wasn't present before?  Maybe the best thing is to press on with the importer since these types of decisions will have to be made and it will become clearer.

Reply:
>  Maybe the best thing is to press on with the importer since these types of decisions will have to be made and it will become clearer.

Agreed, I'll include the importer in this PR.